Neos.ContentRepository.BehavioralTests/Tests/Behavior/Features/EventSourced/Migration/AddDimensionShineThrough.feature

```diff
@@ -74,7 +74,7 @@ Feature: Add Dimension Specialization
     # the original content stream has not been touched
     When I am in workspace "live"
     And I am in dimension space point {"language": "de"}
-    Then I expect node aggregate identifier "sir-david-nodenborough" to lead to node cs-identifier;sir-david-nodenborough;{"language": "de"}
+    Then I get the node with id "sir-david-nodenborough"
```

Comment:
i cannot really judge this. 
And by implementation it seems no error is thrown if the node doesnt exist.

cc mr testsuite aka @nezaniel how should the test suite evolve?

Reply:
> And by implementation it seems no error is thrown if the node doesnt exist.
But we have expectations on this node below. So in case the node doesn't exist, it would fail on the next expectation.

Reply:
I'd rather not reduce test precision here; thi new test would also succeed If we copied the whole node via variation and not just the Edge for fallback

Reply:
The issue is, that we do not know the contentstreamid here.

Reply:
As I said, not in this case. The node migration will create a new workspace with a new contentstreamId, which we don't know, as they is created inside of the node migration.

Reply:
I don't agree here. This test is to be meant as kind of black box test. I have some input and get some result, which I like to verify. I don't want to know what happens inside of the feature. Otherwise a unit test would be more suiteable. And IMHO adding things, just for testing is a bad pattern.

Reply:
> And IMHO adding things, just for testing is a bad pattern.

hmm it depends. For more deterministic behaviour we have many places where its `ContentStreamId::create()` by default but allow to override it. (Like also the predefined node ids for tethered childnodes) ... not sure whats best here though.

Reply:
In general you're right, but that is not how the test suite is designed.
We always test
* Given some commands
* When interesting new command
* Then events
* And projection
There are no lower level tests, especially for this one. Thus the Behat tests must check even the internals. Higher level tests (e.g. for the Neos Workspace model) would not have to do this, but for this to work we have to do it here.

Reply:
okay i though of this again, i think its fine to loose testing precision regarding operating on the workspace instead of the content stream. Buuut we should still check that the node we found is originates from the current dimension were in.


>  When I am in workspace "live"
>  And I am in dimension space point {"language": "de"}
>  Then I expect the node with id "sir-david-nodenborough" to originate from the current dimension space


And the implementation:

```diff
    /**
      * @Then /^I expect the node with id "([^"]*)" to originate from the current dimension space$/
      * @param string $id
      * @throws \Exception
      */
     public function iGetTheNodeWithId(string $id): void
     {
         $nodeAggregateId = NodeAggregateId::fromString($id);
         $this->initializeCurrentNodeFromContentSubgraph(function (ContentSubgraphInterface $subgraph) use ($nodeAggregateId) {
             $node = $subgraph->findNodeById($nodeAggregateId);
+             assert($node->originDimensionSpacePoint->toDimensionSpacePoint()->equals($currentDSP));
             return $node;
         });
     }
```

Reply:
Isn't the whole concept of adding shine-throughs about a node being visible, but _not_ originating in the current DSP?

Also do we really have to introduce another way of comparing nodes here? Streamlining this case with the rest of the test suite shouldn't be that hard.
Plus assigning a deterministic content stream ID is nothing we need in userland yet, but I'd definitivly not rule out that it might be helpful beside the tests.